### DISCUSSION TITLE
kubernetes-handbook/auth/network-policy/

### DISCUSSION THREAD

**Discussion by giscus** [rootsongjc/cloud-native-library#109] [Announcements]
# kubernetes-handbook/auth/network-policy/

网络策略说明一组 Pod 之间是如何被允许互相通信，以及如何与其它网络 Endpoint 进行通信。 NetworkPolicy 资源使用标签来选择 Pod，并定义了一些规则，这些规则指明允许什么流量进入到选中的 Pod 上。关于 Network Policy 的详细用法请参考 Kubernetes 官网。 Network Policy 的

https://lib.jimmysong.io/kubernetes-handbook/auth/network-policy/

**Comment by yixiubest** [comment]
瞎翻译
So, the example NetworkPolicy:

1.isolates role=db pods in the default namespace for both ingress and egress traffic (if they weren't already isolated)

2.(Ingress rules) allows connections to all pods in the default namespace with the label role=db on TCP port 6379 from:

    any pod in the default namespace with the label role=frontend
    any pod in a namespace with the label project=myproject
    IP addresses in the ranges 172.17.0.0–172.17.0.255 and 172.17.2.0–172.17.255.255 (ie, all of 172.17.0.0/16 except 
    172.17.1.0/24)
3.(Egress rules) allows connections from any pod in the default namespace with the label role=db to CIDR 10.0.0.0/24 on TCP port 5978

**Reply by rootsongjc** [reply]
非常感谢你的反馈和指正。我之前翻译了这段内容，但时间距今已经很久了，后来原文似乎有所修改，我没有及时更新翻译，导致出现了错误。我深感抱歉，我会立刻修正这个错误。
